query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/converters/PinotFunctionConverter.java

```diff
@@ -0,0 +1,130 @@
+package org.hypertrace.core.query.service.pinot.converters;
+
+import static org.hypertrace.core.query.service.QueryFunctionConstants.QUERY_FUNCTION_COUNT;
+import static org.hypertrace.core.query.service.QueryFunctionConstants.QUERY_FUNCTION_PERCENTILE;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Function;
+import org.hypertrace.core.query.service.api.LiteralConstant;
+import org.hypertrace.core.query.service.api.Value;
+import org.hypertrace.core.query.service.api.ValueType;
+
+public class PinotFunctionConverter {
```

Comment:
Nice.. Thanks for cleaning this up!

Reply:
No problem. Once this is up and running, we can do some clean up on the gateway side to strip the conversion there since this handles it now.